bcgov/GDX-Analytics-URL-Shortener
# Gdxdsd 6231 refine url table

This PR makes changes to the following files:

**'backend/routes.js'
'frontend/url-shortener/src/components/navigation.vue'
'frontend/url-shortener/src/router/index.ts'**
- changes the '/url-table' route to '/urls'

**'backend/urlTable.js'**
- sorts the retrieved list of URLs from the database by 'createdTime'

**'frontend/url-shortener/src/components/ShortUrl.vue'
'frontend/url-shortener/src/views/UrlSummary.vue'**
- changes 'Created Time' to 'Created On'
- changes 'Custom ID' to 'Link #'
- changes 'Shortened URL' to 'Short URL'

**'frontend/url-shortener/src/components/UrlTable.vue'**
- changes 'Created Time' to 'Created On'
- changes the default sort view for the table
- changes 'Custom ID' to 'Link #'
- changes 'Shortened URL' to 'Short URL'
- makes Short URL clickable and adds a Copy URL icon
- adds a CopyToClipboard function
- eliminates wrapping and indicates text truncation in Target URL and Notes section 
- makes Target URL and Notes section expandable by clicking on the cells
- adds necessary styles

To test the PR:
1. Copy the repo and switch to the relevant branch

```
git clone https://github.com/bcgov/GDX-Analytics-URL-Shortener.git
git checkout GDXDSD-6231-Refine-URL-Table
```
2. Follow the steps in [backend/README.md ](https://github.com/bcgov/GDX-Analytics-URL-Shortener/blob/main/backend/README.md)to configure and start the URL-S backend app

3. Follow the steps in [frontend/README.md](https://github.com/bcgov/GDX-Analytics-URL-Shortener/blob/main/frontend/README.md) to configure and start the URL-S frontend app

4. Browse to 'http://localhost:5173/shorten' and generate several short URLs. Use long Target URLs and optional notes.

5. Go to 'http://localhost:5173/urls' and check if

- the column names are: Link #, Short URL, Target URL, Expiry Date, Created By, Created On, Notes
- the table is sorted initially by the 'Created On' column
- the Short URL field displays a link and the copy icon functions as expected
- oversized Target URL and Notes cells are displayed with (...) and expand when clicked
